src/components/application_manager/src/commands/mobile/register_app_interface_request.cc

```diff
@@ -590,9 +590,9 @@ void RegisterAppInterfaceRequest::SendRegisterAppInterfaceResponseToMobile() {
                                             application->mac_address());
   }
 
+  SendResponse(true, result_code, add_info.c_str(), &response_params);
   SendOnAppRegisteredNotificationToHMI(
       *(application.get()), resumption, need_restore_vr);
-  SendResponse(true, result_code, add_info.c_str(), &response_params);
```

Comment:
@anosach-luxoft After looking through the Confluence project, this fix does appear to break the sequence of messages which are sent after receiving a RegisterAppInterface request:

> The order of requests/responses/notifications during application registering must be the following:
1.app->SDL: RegisterAppInterface (policy_appID, parameters)
2.SDL->HMI: OnAppRegistered (hmi_appID, params)
3.SDL->app: RegisterAppInterface_response (<applicable resultCode>, success:true)
4. SDL->app: OnHMIStatus(hmiLevel, audioStreamingState, systemContext)
5. SDL->app: OnPermissionsChange(params)

Is there any specific reason why the HMI needs to be notified of an app registration prior to the mobile device?